crates/storage/src/collections/vec/mod.rs

```diff
@@ -200,6 +200,163 @@ where
         *self.len += 1;
         self.elems.put(last_index, Some(value));
     }
+
+    /// Binary searches this sorted vector for a given element.
```

Comment:
Can you add a comment explaining where this comes from and a link to the original source code? It doesn't need to be part of the doc comments since it would mostly be of interest to devs working on the internals